build_tools/github_actions/runner/config/setup.sh

```diff
@@ -6,29 +6,50 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# Sets up GitHub actions runner services to start and teardown runner.
-# Registers the runner followed by enabling deregister service then 
-# starts the registered runner to accept self-hosted GitHub workflows.
+# Installs and sets up the GitHub actions runner, creating services to start and
+# tear down the runner.
 
-set -euo pipefail
+set -xeuo pipefail
 
-echo "Register the self-hoster runner."
-chmod +x ~runner/config/register.sh
-runuser --user runner ~runner/config/register.sh
+SCRIPT_DIR="$(dirname -- "$( readlink -f -- "$0"; )")";
+source "${SCRIPT_DIR}/functions.sh"
 
-echo "Setup the deregister service."
-cp ~runner/config/github-actions-runner-deregister.service /etc/systemd/system/
-chmod +x ~runner/config/deregister.sh
+echo "Creating tmpfs for runner"
+mkdir /runner-root
+mount -t tmpfs -o size=50g tmpfs /runner-root
```

Comment:
Tested on my local machine, looks like by default it gives 0777 file permission. Do we want to restrict the access a little?

Reply:
I don't think it really matters? The runner has passwordless sudo